src/ethereum/txUtils.ts

```diff
@@ -43,14 +43,22 @@ export namespace txUtils {
    * @return {object} data - Current transaction data. See {@link txUtils#getTransaction}
    */
   export async function waitForCompletion(txId: string): Promise<any> {
+    let retriesOnEmpty = 5 // (5 * TRANSACTION_FETCH_DELAY) seconds
+
     while (true) {
```

Comment:
Since we have a discrete number of amounts to iterate, we could replace the implementation by a for-loop an it would be cleaner

Reply:
Kind of, because the idea is to wait until the transaction ends _or_ it's dropped. I'm thinking this is a better implementation 

```typescript
  export async function waitForCompletion(txId: string, retriesOnEmpty: number = 5): Promise<any> {
    const isDropped = await isTxDropped(txId, retriesOnEmpty)
    if (isDropped) {
      return { hash: txId, status: TRANSACTION_STATUS.failed }
    }

    while (true) {
      const tx = await getTransaction(txId)
      
      if (!isPending(tx) && tx.recepeit) {
        return tx
      }

      await sleep(TRANSACTION_FETCH_DELAY)
    }
  }

  export async function isTxDropped(txId: string, retryAttemps: number = 5) : boolean {
    while (retryAttemps > 0) {
      const tx = await getTransaction(txId)

      if (tx !== null) {
        return false
      }

      retryAttemps -= 1
      await sleep(TRANSACTION_FETCH_DELAY)
    }

    return true
  }
```

what do you think?